lib/streamlit/runtime/media_file_manager.py

```diff
@@ -264,19 +265,33 @@ def add(
         -------
         str
             The url that the frontend can use to fetch the media.
+
+        Raises
+        ------
+        If a filename is passed, any Exception raised when trying to read the
+        file will be re-raised.
         """
+
+        # If we were passed a filename, open and read the file into memory.
+        content: bytes
+        if isinstance(data_or_filename, str):
+            with open(data_or_filename, "rb") as f:
+                content = f.read()
+        else:
+            content = data_or_filename
```

Comment:
We still just read the file directly into memory even when passed by filename. The "delegate file reading to the `MediaFileStorage` interface" functionality will come in a future PR, before this feature is merged to develop!